File: dotnet/src/Core/Models/Entities/Authorizer.cs
using Agience.Core.Models.Entities.Abstract;
using Agience.Core.Models.Enums;
using System.Text.Json.Serialization;

namespace Agience.Core.Models.Entities
{
    public class Authorizer : PublicEntity
    {        

        [JsonPropertyName("client_id")]
        public string? ClientId { get; set; }

        [JsonPropertyName("client_secret")]
        public string? ClientSecret { get; set; } // TODO: SECURITY: Use a key vault

        [JsonPropertyName("redirect_uri")]
        public string RedirectUriPath => $"/oauth/authorizer/{Id}/callback";

        [JsonPropertyName("auth_uri")]
        public string? AuthUri { get; set; }

        [JsonPropertyName("token_uri")]
        public string? TokenUri { get; set; }

        [JsonPropertyName("scopes")]
        public string? Scopes { get; set; }

        [JsonPropertyName("auth_type")]
        public AuthorizationType AuthType { get; set; } = AuthorizationType.Public;

    }
}

File: python/src/core/models/entities/authorizer.py
from typing import Optional
from .agience_entity import AgienceEntity
from .enum import Visibility, AuthorizationType


class Authorizer(AgienceEntity):
    def __init__(self,
                 name: Optional[str] = None,
                 manager_id: Optional[str] = None,
                 client_id: Optional[str] = None,
                 # TODO: SECURITY: Use a key vault
                 client_secret: Optional[str] = None,
                 auth_uri: Optional[str] = None,
                 token_uri: Optional[str] = None,
                 scope: Optional[str] = None,
                 auth_type: Optional[AuthorizationType] = None,
                 visibility: Optional[Visibility] = Visibility.Private,
                 id: str = ""):
        super().__init__(id=id)
        self.name = name
        self.manager_id = manager_id
        self.client_id = client_id
        self.client_secret = client_secret
        self.auth_uri = auth_uri
        self.token_uri = token_uri
        self.scope = scope
        self.auth_type = auth_type
        self.visibility = visibility

    @property
    def redirect_uri(self) -> str:
        # TODO: Build the Redirect URI based on the type of authorizer, in a different class
        # Include the Authority part
        return f"/manage/authorizer/{self.id}/callback"
